react/features/base/connection/functions.js

```diff
@@ -3,23 +3,47 @@
 import { toState } from '../redux';
 
 /**
- * Retrieves a simplified version of the conference/location URL stripped of URL
- * params (i.e. Query/search and hash) which should be used for sending invites.
+ * Retrieves a simplified version of the conference/location URL stripped of URL params (i.e. Query/search and hash)
+ * which should be used for sending invites.
+ * NOTE that the method will throw an error if called too early. That is before the conference is joined or before
+ * the process of joining one has started. This limitation does not apply to the case when called with the URL object
+ * instance. Use {@link isInviteURLReady} to check if it's safe to call the method already.
  *
- * @param {Function|Object} stateOrGetState - The redux state or redux's
- * {@code getState} function.
+ * @param {Function|Object} stateOrGetState - The redux state or redux's {@code getState} function or the URL object
+ * to be stripped.
  * @returns {string}
  */
 export function getInviteURL(stateOrGetState: Function | Object): string {
     const state = toState(stateOrGetState);
-    const locationURL
+    let locationURL
         = state instanceof URL
             ? state
             : state['features/base/connection'].locationURL;
 
+    // If there's no locationURL on the base/connection feature try the base/config where it's set earlier.
+    if (!locationURL) {
+        locationURL = state['features/base/config'].locationURL;
+    }
+
+    if (!locationURL) {
+        throw new Error('Can not get invite URL - the app is not ready');
+    }
+
     return getURLWithoutParams(locationURL).href;
 }
 
+/**
+ * Checks whether or not is safe to call the {@link getInviteURL} method already.
+ *
+ * @param {Function|Object} stateOrGetState - The redux state or redux's {@code getState} function.
+ * @returns {boolean}
+ */
+export function isInviteURLReady(stateOrGetState: Function | Object): boolean {
```

Comment:
I think we should try / catch here instead, because if `getInviteURL` may change and this function would need to be adjusted, I think.

Or getter, just try-catch where `getInviteURL` is used?

Reply:
I disagree with try catching on a regular basis when the exception can be avoided. If this function will need adjustment we're going to see that by exceptions being thrown and not handled. Eventually both are linked by descriptions in the JSdoc, no ?

Reply:
Exceptions are "exceptions" and not a norm

Reply:
We can always go back to the original version where one needs to check for the `undefined` result...

Reply:
Maybe it's my Python bias, leave it as-is then.